NuRadioReco/modules/io/RNO_G/readRNOGDataMattak.py

```diff
@@ -640,6 +646,52 @@ def get_events_information(self, keys=["station", "run", "eventNumber"]):
 
         return self._events_information
 
+    def get_waveforms(self, remove_block_offsets=False):
+        """ Return waveforms of events passing the selectors which may have been specified
+
+        .. Warning:: The function to remove the block offsets is not the same as the one used in
+            the regular interface of this class.
+
+        Parameters
+        ----------
+
+        remove_block_offsets: bool
+            If True, use a simple function `_baseline_correction` (defined at the top if this file)
+            to remove the block offset. (Default: False)
+
+        Returns
+        -------
+
+        wfs: np.array
+            Waveforms of all "selected" events. The wavefroms are either calibrated or not
+            based on the class config.
+        """
+
+        if self._events_waveforms is None:
+            self._events_waveforms = []
```

Comment:
I can kind of see the appeal of 'caching' the result in the class? But this does compound the risk of using a lot of memory (see also my earlier comment), and in the current implementation it means any arguments passed to `get_waveforms` are ignored after the initial call (e.g. if I first get the waveforms without block offset removal then the second time I call the function I will get them without block offset removal regardless of what argument I pass to the function).

Reply:
In the workflow I have been using so far it came in handy but you have good arguments.